test/performance/performance.go

```diff
@@ -152,3 +158,69 @@ func sanitizedURL(endpoint string) string {
 	}
 	return endpoint
 }
+
+// DeploymentStatus is a struct that wraps the status of a deployment.
+type DeploymentStatus struct {
+	DesiredReplicas int32
+	ReadyReplicas   int32
+}
+
+// FetchDeploymentStatus creates a channel that can return the up-to-date DeploymentStatus periodically.
+func FetchDeploymentStatus(
+	ctx context.Context, namespace string, selector labels.Selector, ticker *time.Ticker,
```

Comment:
Can we have this take a duration and start its own ticker?

Reply:
Done.